### PR TITLE
A new helper method in Decorator class to resolve types for type references decorators

### PR DESCRIPTION
**Pull request by RINO-GAELICO**
…r type references decorators. Failed tests with Error message: Internal error: local types are not yet initialized. Do not try to resolve types inside . The error is thrown by ModelFile.getLocalType method. Inside the _resolveArguemntType method the call that lead to the error is modelFile.getType(argument.type.name)

# Closes #<#797>

As requested by @sanketshevkar I created this draft pull request to allow him to dig up into this.
### Changes

- A new helper method in Decorator class to resolve types for type references decorators >
- Failed tests with Error message: Internal error: local types are not yet initialized. Do not try to resolve types inside . The error is thrown by ModelFile.getLocalType method. Inside the _resolveArguemntType method the call that lead to the error is modelFile.getType(argument.type.name)

### Flags

- <ONE>
- <TWO>

### Related Issues
- Issue #798 
- Pull Request #<NUMBER>

### Author Checklist
- [ ] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

File: packages/concerto-vocabulary/test/org.acme@1.0.0.cto [hunk 1]
```diff
@@ -9,7 +9,7 @@ enum Color {
 scalar SSN extends String default="000-00-0000"
 scalar VIN extends String
 
-asset Vehicle identified by vin {
+asset Vehicle identified {
```
Comment (sanketshevkar) [inline packages/concerto-vocabulary/test/org.acme@1.0.0.cto:12]: Why is this changed?

File: packages/concerto-core/src/introspect/decorator.js [hunk 2]
```diff
@@ -106,6 +111,42 @@ class Decorator extends ModelElement {
...
+     * @param {Object} argument - the decorator argument // Idt we have argument as any defined type
+     * @return {ClassDeclaration} the type of the decorator argument // similar to [this](https://github.com/accordproject/concerto/blob/0d7f108d4961f87782436dd689639ce0f35e97f8/packages/concerto-core/lib/introspect/classdeclaration.js#L155) method
+     */
+    _resolveArguemntType(argument) {
+
+        let classDecl = null;
+
+        const modelFile = this.parent.getModelFile();
+
+        if (modelFile.isImportedType(argument.type.name)) {
```
Comment (sanketshevkar) [inline packages/concerto-core/src/introspect/decorator.js:135]: I think just `modelFile.getType()` should be able to give you classDeclacarations that are imported or present within the namespace.
https://github.com/accordproject/concerto/blob/3dddd1ec5c41be47d1e2f3070de5d979afa8e23f/packages/concerto-core/lib/introspect/modelfile.js#L409

**Review by sanketshevkar — CHANGES_REQUESTED** [review]

**Comment by sanketshevkar** [comment]
Regarding this [comment](https://github.com/accordproject/concerto/issues/797#issuecomment-1995397586) from the issue.

Yes, you're findings seem correct to me. For this, I think we can use this in the `validate` [method](https://github.com/accordproject/concerto/blob/3dddd1ec5c41be47d1e2f3070de5d979afa8e23f/packages/concerto-core/lib/introspect/decorated.js#L112).

@mttrbrts @dselman will this be a good approach?

File: packages/concerto-vocabulary/test/org.acme@1.0.0.cto (hunk 1 above)
Comment (RINO-GAELICO) [reply packages/concerto-vocabulary/test/org.acme@1.0.0.cto:12]: No specific reasons, I was working on another feature #776 and I decided to test it that way.
I will change it back.
